transformer_engine/pytorch/csrc/ts_fp8_op.cpp

```diff
@@ -100,6 +104,9 @@ at::Tensor te_gemm_ts(at::Tensor A,
   if (B_scale_inverse.numel())
     B_scale_inverse_arg = B_scale_inverse[B_fp8_tensor];
 
+  at::Tensor D_scale_arg = D_scale.clone();
+  at::Tensor D_amax_arg = D_amax.clone();
```

Comment:
Why do you need to clone them?

Reply:
I just copied the usage for other scale factors such as
at::Tensor A_scale_inverse_arg = A_scale_inverse.clone();

Is it unnecessary?

Reply:
@ksivaman could you comment on that?

Reply:
Yes this shouldn't be cloned for 2 reasons:
1. Unnecessary malloc,
2. and more importantly, we want to populate the original amax tensor received in the arguments, which would remain unchanged here.

Reply:
Thanks. I'll make the change.

Reply:
@ksivaman That begs the question why are there the other clones for scale inverses? I believe those are unnecessary as well and create additional (albeit pretty small)D2D copies before every FP8 gemm.

Reply:
Yes, I believe all 7 clones in `transformer_engine/pytorch/csrc/ts_fp8_op.cpp` can be removed. The scale inverse clones also seem unused? @asfiyab-nvidia Could you comment on why these were added to begin with?

Reply:
@ksivaman These aren't necessary. I'll create a PR with a fix shortly. Thanks for pointing it out